src/main/java/seedu/address/storage/XmlAddressBookStorage.java

```diff
@@ -74,4 +85,63 @@ public void saveAddressBook(ReadOnlyAddressBook addressBook, String filePath) th
         XmlFileStorage.saveDataToFile(file, new XmlSerializableAddressBook(addressBook));
     }
 
+    @Override
+    public void backupAddressBook() throws IOException {
+        try {
+            Optional<ReadOnlyAddressBook> addressBookOptional;
+            ReadOnlyAddressBook initialData;
+            addressBookOptional = readAddressBook();
+            if (!addressBookOptional.isPresent()) {
+                logger.info("No backup will be made as data file does not exist");
+            } else {
+                initialData = addressBookOptional.orElseGet(SampleDataUtil::getSampleAddressBook);
+                saveAddressBook(initialData, getBackupAddressBookFilePath());
+            }
+        } catch (DataConversionException e) {
+            logger.info("No backup will be made due to data file being in incorrect format");
+        } catch (IOException e) {
+            logger.info("No backup will be made due to problems while reading from file");
+        }
+    }
+
+    @Override
+    public ReadOnlyAddressBook getBestAvailableAddressBook() {
+        ReadOnlyAddressBook initialData = new AddressBook();
+        boolean dataFileIsOkay = false;
+        Optional<ReadOnlyAddressBook> addressBookOptional;
+
+        try {
+            addressBookOptional = readAddressBook();
+            if (addressBookOptional.isPresent()) {
+                dataFileIsOkay = true;
+            } else {
+                logger.info("Data file not found");
+            }
+            initialData = addressBookOptional.orElseGet(SampleDataUtil::getSampleAddressBook);
+        } catch (DataConversionException e) {
+            logger.warning("Data file not in the correct format");
+        } catch (IOException e) {
+            logger.warning("Problem while reading from the file");
+        }
+
+        if (!dataFileIsOkay) {
```

Comment:
[Coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html) violation! One more more of these problems detected:
1. name doesn't follow naming convention

Reply:
Noted. Necessary changes have been made by the group.